mobile/src/slices/auth/actions.ts

```diff
@@ -5,20 +5,32 @@ import { type AsyncThunkConfig } from '#libs/types/types';
 import {
   type UserAuthResponseDto,
   type UserSignUpRequestDto,
+  type UserSignUpResponseDto,
 } from '#packages/users/users';
 
 import { name as sliceName } from './auth.slice';
 
 const signUp = createAsyncThunk<
-  UserAuthResponseDto,
+  UserSignUpResponseDto,
   UserSignUpRequestDto,
   AsyncThunkConfig
->(`${sliceName}/sign-up`, async (signUpPayload, { extra }) => {
+>(`${sliceName}/sign-up`, async (registerPayload, { extra }) => {
   const { authApi, storage } = extra;
-  const { user, token } = await authApi.signUp(signUpPayload);
-  await storage.set(StorageKey.TOKEN, token);
+  const response = await authApi.signUp(registerPayload);
+
+  await storage.set(StorageKey.TOKEN, response.token);
+
+  return response;
```

Comment:
```suggestion
  const { user, token } = await authApi.signUp(registerPayload);

  await storage.set(StorageKey.TOKEN, token);

  return user;
```

Reply:
i get an error here >(`${sliceName}/sign-up`, async (signUpPayload, { extra }) => { when i try to do so

Reply:
You need to replace UserSignUpResponseDto with UserAuthResponseDto in first type argument of this action